### PR TITLE
Docs: PyPI Setup & Deploy #67

### PR DESCRIPTION
**Pull request by yasheela-alla**
### Related Issue

Closes: #67 

### Description

This documentation provides a step-by-step guide on how to set up and deploy a Python package to PyPI (Python Package Index). 

### Checklist

- [ ] I have crafted the documentation.
- [ ] I have followed the contribution guidelines
- [ ] I have ensured my changes don't generate any new warnings or errors.
- [ ] I have resolved all merge conflicts.

### CODE REVIEW

**Comment by yasheela-alla** [comment]
Hi @this-is-yaash, Please review the Doc.

**Comment by jindalpriyanshu101** [comment]
Hey @yasheela-alla, can you correctly link the issue number which this PR intend to fix? as you've linked the discussion instead of Issue

File: PyPi_Setup.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    └── test_module.py    # Example test file
+```
+
+### Explanation of Key Files
+
+- **setup.py**: Contains package information and dependencies.
+- **README.md**: Provides information about the package.
+- **my_package/**: Directory containing the actual Python code.
+- **tests/**: Directory for unit testing your package.
+
```
Comment (this-is-yaash) [inline PyPi_Setup.md:65]: Use the actual repository project structure.
It seems totally unrelated! Doesn't make sense!

Comment (this-is-yaash) [reply PyPi_Setup.md:65]: Use same naming conventions as well, 
But at the end, add a saying that feel free change the names

and inside the `stash` dir contains the features, mention that these filenames are aliases. feel free to change it to meet your needs!

File: PyPi_Setup.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+setup(
+    name='your_package_name',
+    version='0.1',
+    packages=find_packages(),
+    install_requires=[
+        # Add any package dependencies here
+    ],
+    # You can also add metadata like author, license, etc.
+)
+```
```
Comment (this-is-yaash) [inline PyPi_Setup.md:84]: Use the `setup.py` in this repo for reference. Copy-pasting is also enf!

File: PyPi_Setup.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+## 6. Uploading to PyPI
+
+### Detailed Steps for Deployment
+
+1. **Create a PyPI Account**:
+   - Go to PyPI and create an account if you don’t have one.
+   - After logging in, generate an API token via your account settings.
+
+2. **Upload the Package Using Twine**:
+   - Use your API token for authentication:
```
Comment (this-is-yaash) [inline PyPi_Setup.md:105]: Elaborate with steps, it's going to be tedious process if documented with lil info!

**Review by this-is-yaash — COMMENTED** [review]
Please make the necessary updates as per the review

File: PyPi_Setup.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
+# PyPi setup 
+
+## 1. Introduction
+
+This documentation provides a comprehensive guide on how to set up and deploy a Python package to the Python Package Index (PyPI). The package is designed to facilitate the sharing of Python projects, enabling users to install it easily using pip.
+
+### Purpose
+
+The objectives of this guide are to help users:
```
Comment (ShashwatAgrawal20) [inline PyPi_Setup.md:9]: Can you change this to first person, idk is sounds a little wired while mentoning users in the docs, it should be something like 
```suggestion
The objectives of this guide is to help you with:
```

File: PyPi_Setup.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
+# PyPi setup 
+
+## 1. Introduction
+
+This documentation provides a comprehensive guide on how to set up and deploy a Python package to the Python Package Index (PyPI). The package is designed to facilitate the sharing of Python projects, enabling users to install it easily using pip.
```
Comment (ShashwatAgrawal20) [inline PyPi_Setup.md:5]: I feel like this can be better, it doesn't sound that good with that `This documentation` it should be something in the first person I feel. CC @this-is-yaash

**Review by ShashwatAgrawal20 — CHANGES_REQUESTED** [review]
